akka-stream/src/main/scala/akka/stream/scaladsl/Queue.scala

```diff
@@ -6,6 +6,17 @@ package akka.stream.scaladsl
 import scala.concurrent.Future
 import akka.Done
 import akka.stream.QueueOfferResult
+import akka.stream.stage.GraphStageLogic
+
+import scala.util.control.NoStackTrace
+
+/**
+ * This exception signals that materialized value is already detached from stream. This usually happens
+ * when stream is completed and an ActorSystem is shut down while materialized object is still available.
+ */
+final class StreamDetachedException
```

Comment:
shouldn't this be in akka.stream package since it may be exposed in javadsl also, similar to AbruptStageTerminationException